posts/_markdown/vqa_in_qutip.md

```diff
@@ -0,0 +1,106 @@
+---
+title: Implementing a Variational Quantum Algorithms module in QuTiP
+date: 03 February 2022
+author: Ben Braham
+---
+## About me 
+
+I'm an undergraduate student at the University of Sydney, studying Physics and Computer Science. I've worked on a few quantum computing projects in the past and was really excited to be awarded a microgrant by the Unitary Fund to work on this project.
+
+## Background
+
+
+Many groundbreaking quantum algorithms (e.g. Shor's, Grover's) require quantum computers with a large number of qubits and high fault-tolerance to produce useful output for non-trivial problem instances. While this is likely some time away, there exist algorithms that aim to leverage the current power of noisy intermediate--scale quantum (NISQ) systems. Variational Quantum Algorithms (VQAs) are a hybrid classical-quantum type of algorithm in which parameterized quantum circuits (PQCs) are optimized, with an objective function being evaluated on quantum hardware and the parameter update occurring on a classical machine.
+
+One current area of interest in VQAs is that of the "initialization" for the algorithm. Problems are described by a quantum circuit, with its structure based on a circuit ansatz - usually inspired by the problem one wishes to solve. The performance of the algorithm is heavily influenced by the circuit structure, as well as the initial parameters of the circuit. It is therefore worthwhile to have the tools to test VQA formulations on a classical simulation of a quantum system, before expending time and resources to apply it on real quantum hardware.
+
+
+`QuTiP` is a open-source `python` library for simulating quantum systems. In this project, I designed and wrote an implementation of a `VQA` module for `QuTiP` which allows users to define and evaluate the effectiveness of VQAs on their local machines.
+
+## Module Overview
+
+Most quantum algorithms deal with quantum circuits, which are made up of quantum gates. In `QuTiP`, there exist classes --- `QubitCircuit` and `Gate` --- which simulate the dynamics of quantum circuits.
+
+For VQAs, we now want these gates to be parameterized. In addition to the gate parameters, we want initialization options, which are fixed throughout the optimization process, to be easily defined in one object. For these purposes, we define two new classes, `VQA_Block` and `VQA`, which are abstractions on `Gate` and `QubitCircuit`, respectively.
+
+### 1. The VQA_Block class
+
+The `VQA_Block` class encapsulates one component of our circuit and can store the action of one or more quantum gates. It can hold parameters, or keep note of how many free parameters it needs as input to be translated into a gate on the quantum circuit.
```

Comment:
```suggestion
The `VQABlock` class encapsulates one component of our circuit and can store the action of one or more quantum gates. It can hold parameters, or keep note of how many free parameters it needs as input to be translated into a gate on the quantum circuit.
```